84-sarcastic-fringehead/technical_specification.md

```diff
@@ -0,0 +1,238 @@
+
+## Scope
+### Outline:
+This epic includes all work required to bring the remaining file services into line with the new file upload concept. The first portion of work for the file services was executed under [Lynx Boreal](../76-lynx-boreal/technical_specification.md), and there was also a subsequent portion of work for the GHGA Connector which was carried out according to [Hedgehog Seahorse](../80-hedgehog-seahorse/technical_specification.md). When this epic is finished, all *backend* modifications required for the new upload concept to be realized will be complete. Frontend changes are *not* included in this epic, however, so more work will be required to bring the Data Portal up to speed.
+As for the work to be completed within this epic, the services affected include the File Ingest Service (FIS), Internal File Registry Service (IFRS), the Well-Known Value Service (WKVS), the UCS, the ghga-event-schemas library, and a new service called the Data Hub File Service (DHFS). Additionally, if it is discovered during implementation that further changes need to be made to other services *beyond what is described this epic*, e.g. to add an endpoint parameter, then tickets will be added ad-hoc and associated with this epic.
+
+In Lynx Boreal, the Upload Controller Service (UCS) was rewritten, the Upload Orchestration Service (UOS) was implemented for the first time, the Claims Repository Service (CRS) was updated to manage permissions for Research Data Upload Boxes, and the Work Package Service (WPS) was updated to manage upload-type work packages. Taken together, these changes create the operational framework for remote file upload, but only to the point of initial ingest. In order to comply with our file upload concept, we still need to decrypt the uploaded file, verify the integrity via checksum comparison, re-encrypt the file with a new file secret (securely stored in the Encryption Key Store Service, or EKSS), and move the file to a permanent storage bucket registered with the IFRS in what we call "archival".
+
+
+### Included/Required:
+All work described in the Additional Implementation Details section below is required.
+
+### Optional:
+Add email notifications for important events related to archival. This could include, for example, a notification conveying that all files in a Research Data Upload Box have been successfully archived, or that there was a problem with file XYZ during interrogation. To prevent scope creep, this should *probably* be done in another epic, but we should keep that potential requirement in mind during development.
+
+### Not included:
+Data Portal updates or any upcoming metadata-related services. This is purely for file upload.
+
+## User Journeys (optional)
+
+All user journeys are already detailed in Lynx Boreal. The operations added in this epic will occur automatically without further action required on the part of either the user or GHGA personnel. 
+
+## API Definitions:
+
+### RESTful/Synchronous:
```

Comment:
Need a refresher here, but is the information returned by these endpoints scoped based on claims in an authentication token or how is the visible/returnable information determined?

Reply:
I was thinking about whether the endpoints return all information for a storage alias unconditionally or if there's information in the token to  scope down the returned information to the corresponding research upload box.